lib/spack/docs/conf.py

```diff
@@ -81,7 +81,7 @@
 for filename in glob('*rst'):
     with open(filename) as f:
         for line in f:
-            match = re.match(r'.. _(spack-[^:]*)', line)
+            match = re.match(r'.. _(spack-[a-z]{2,}):', line)
```

Comment:
@tgamblin My last push hid your comments, but how do you feel about this hack? It skips command+flag sections like `_spack-edit-f` and `_spack-p`. I could also rename these sections to `_cmd-spack-*` to make it more explicit, which would prevent bugs like how `spack-cray` was showing up before. If you want me to go down that route, I'll probably wait until after #1676 is merged so that @citibeth doesn't have to rebase again.